Add new artifact: Bloodletter

RikerW opened the pull request:
* unaligned (iron) morning star
* +1d8 to hit and +1d12 blood damage
  * Deals extra bonus blood damage at lower health. At under 50% of maximum health, deals an additional +1d12 for a total of +2d12. At under 25% of maximum health, deals an additional +1d12 for a total of +3d12.
* When invoked, self-impales (dealing damage equal to 20% of your max hp, can be lethal) to coat in blood. This lasts for 20 + 2z9 turns. While coated in blood:
  * Appears as "sanguine" in your inventory. 
  * Damage dealt is also treated as energy damage, in addition to being blunt and piercing.
  * Deals an extra flat +defender level damage to monsters with blood. 
  * Hitting high level monsters during this time can decrease the amount of time remaining on the blood coating. It subtracts `defender_level/10 - rn2(5)` (rounded down) from the time remaining, and never adds more time.
  * Can be applied to slam down, releasing the tainted blood. This ends the active effect but causes a 6d6 rotten blood explosion. The explosion only hurts creatures with blood, and also deals a flat +defender level damage to affected creatures.

RikerW committed:
Add new artifact: Bloodletter

* unaligned (iron) mace
* +1d8 to hit and +1d12 blood damage
** not technically rotten blood - it uses AD_BLUD, but only checks & does standard damage depending on has_blood_mon
* When invoked, self-impales (dealing damage equal to 20% of your max hp, can be lethal) to coat in blood.
** While coated in blood, appears as "sanguine" and deals an extra d(defender level) damage to monsters with blood
* At <1/2 hp, deals double damage. At <1/4 hp, deals triple damage (replaces the double, obviously, so it's not 6x damage)

RikerW committed:
Adjust bonus damage

It's now a flat +mlev bonus, but reduces the remaining time by (mlev/10) - rn2(5) (floored, never increases duration). That means that you chould hypothetically get an average of 4 or so hits on Demogorgon, each dealing +108 flat in addition to whatever base damage it deals. Stupid? Yeah, quite possibly. Reasonably balanced? Dunno. I'm expecting it to be quite lethal at killing demon lords if you bother to ward & prep beforehand, but not otherwise. Might revisit the initial damage and increase it to 40% or something like that.

RikerW committed:
Change base item type, inspect correctly

NeroOneTrueKing commented:
~~Corrections:~~
* When under 50% hp, deals an additional 1d12 damage to creatures with blood (stacking with previous 1d12)
* When under 25% hp, deals an additional 1d12 damage to creatures with blood (stacking with previous 2d12)
* * Bloodletter does not multiply its base damage or enchantment.
* Invoke duration is `20 + [0-9] + [0-9]` turns; not `20+2d10`.

Edit: main description has been updated with these corrections.

RikerW committed:
Adjust bloodletter somewhat

notable changes:
1. apply to do the AoE, not invoke. invoke is purely activating it
2. applying to do the AoE will END the active period, but not reset invoke time or anything
3. shouldn't crash on a 0 lvl monster anymore
